samcli/lib/build/app_builder.py

```diff
@@ -193,20 +163,44 @@ def _build_function(self, function_name, codeuri, runtime):
             if self._container_manager:
                 build_method = self._build_function_on_container
 
+            options = ApplicationBuilder._get_build_options(config.language, handler)
+
             return build_method(config,
                                 code_dir,
                                 artifacts_dir,
                                 scratch_dir,
                                 manifest_path,
-                                runtime)
+                                runtime,
+                                options)
+
+    @staticmethod
+    def _get_build_options(language, handler):
+        """
+
+        Parameters
+        ----------
+        language str
+            Language of the runtime
+        handler str
+            Handler value of the Lambda Function Resource
+        Returns
+        -------
+        dict
+            Dictionary that represents the options to pass to the builder workflow or None if options are not needed
+        """
+        options = None
+        if language == 'go':
+            options = {'handler': handler}
```

Comment:
hang on. Instead of handler, can we call this option the name of executable? As much as possible, I want to avoid builder specific options. Would this be generic enough? Thoughts?

Reply:
We can make this more generic (will take a PR to aws-lambda-builders). What would you like to call this? `executable_name`?

Reply:
Can we call this something like "artifact_binary_name" or something that explicitly says this is built artifacts?